engine/src/main/java/org/pentaho/di/repository/RepositoryConnectionUtils.java

```diff
@@ -0,0 +1,179 @@
+ */
+
+package org.pentaho.di.repository;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.exception.KettleSecurityException;
+import org.pentaho.di.core.logging.LoggingBuffer;
+import org.pentaho.di.core.plugins.PluginRegistry;
+import org.pentaho.di.core.plugins.RepositoryPluginType;
+import org.pentaho.di.i18n.BaseMessages;
+import org.pentaho.di.repository.RepositoriesMeta;
+import org.pentaho.di.repository.RepositoryMeta;
+import org.pentaho.platform.api.engine.ActionExecutionException;
+import org.pentaho.platform.api.engine.IAuthorizationPolicy;
+import org.pentaho.di.repository.Repository;
+
+import java.io.ByteArrayInputStream;
+
+/**
+ *  Provides a utility method to connect to repository and return the repository object
+ */
+public class RepositoryConnectionUtils {
+  private static final Log logger = LogFactory.getLog( RepositoryConnectionUtils.class );
+  private static final String SINGLE_DI_SERVER_INSTANCE = "singleDiServerInstance";
+
+
+  /**
+   * Connects to the PDI repository. If the  system settings "SINGLE_DI_SERVER_INSTANCE" is set, it will connect to
+   * repository using the embedded repository.xml otherwise it will connect to repository with the given name.
+   *
+   * @param  String RepositoryName
+   * @param LoggingBuffer
+   * @return  Repository
+   * @throws KettleException
+   * @throws KettleSecurityException
+   * @throws ActionExecutionException
+   */
+  public static Repository connectToRepository( final String repositoryName, boolean isSingleDiServerInstance,
+             IAuthorizationPolicy authorizationPolicy, String userName,
+             String fullyQualifiedServerUrl, LoggingBuffer logBuffer ) throws KettleSecurityException, KettleException,
+              ActionExecutionException {
+
+    if ( !hasExecutePermission( authorizationPolicy ) ) {
+      throw new IllegalStateException(
+        BaseMessages.getString( "RepositoryConnectionUtils.ERROR_0001_NO_PERMISSION_TO_EXECUTE" ) ); //$NON-NLS-1$
+    }
+
+    if ( logger.isDebugEnabled() ) {
+      logger.debug( BaseMessages.getString( "RepositoryConnectionUtils.DEBUG_META_REPOSITORY" ) ); //$NON-NLS-1$
+    }
+
+    RepositoriesMeta repositoriesMeta = new RepositoriesMeta();
+
+    if ( logger.isDebugEnabled() ) {
+      logger.debug( BaseMessages.getString( "RepositoryConnectionUtils.DEBUG_POPULATING_META" ) ); //$NON-NLS-1$
+    }
+
+    try {
+      if ( isSingleDiServerInstance ) {
+        if ( logger.isDebugEnabled() ) {
+          logger.debug( "singleDiServerInstance=true, loading default repository" ); //$NON-NLS-1$
+        }
+
+        // Get the embedded repository xml
+        String repositoriesXml = getEmbeddedDefaultRepositoryXml( fullyQualifiedServerUrl );
+
+        ByteArrayInputStream sbis = new ByteArrayInputStream( repositoriesXml.getBytes( "UTF8" ) );
```

Comment:
See if we can use https://github.com/pentaho/pentaho-kettle/blob/master/core/src/main/java/org/pentaho/di/core/Const.java#L393